src/learner.cc

```diff
@@ -585,8 +585,9 @@ class LearnerImpl : public Learner {
         generic_param_.n_gpus = 1;
       }
       if (generic_param_.n_gpus != 1) {
```

Comment:
Not needed for this PR.  But after removing `n_gpus`, the `UseGPU` function should follow ;-).